modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/js/OAuth2Client.ts

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+// @ts-nocheck
+
+import pkceChallenge from 'pkce-challenge';
+
+class OAuth2Client {
+	authorizeURL: any;
+	clientId: any;
+	encodedRedirectURL: string;
+	homePageURL: any;
+	redirectURIs: any[];
+	tokenURL: any;
```

Comment:
All of these `any` types are not ideal. We should scope those to a specific type.

Reply:
fixed